### PR TITLE
Develop <> Master Sync

### PR DESCRIPTION
**Pull request by racerxdl**
- New repository style
- New Mode: SINGLE KEY
- Fix EOF on no public key available for checking
- Fix SaveToDisk KeyBackend trying to load metadata #28 
- Improve DeepSource analysis (by supressing errors that were intentional)

Docker cloud will not be able to build until we merge, since I need to manually change the Dockerfile path in Docker Cloud.

### CODE REVIEW

File: cmd/cli/cmdimport.go [hunk 1]
```diff
@@ -62,35 +61,35 @@ func ImportKey(filename, keyPassword string, keyPasswordFd int) {
 		keyPassword = strings.Trim(string(d), "\n\r")
 	}
 
-	fps, _ := remote_signer.GetFingerPrintsFromKey(string(data))
+	fps, _ := tools.GetFingerPrintsFromKey(string(data))
```
Comment (silviolleite) [inline cmd/cli/cmdimport.go:64]: No error handling. ;)

File: cmd/cli/generateKey.go [hunk 2]
```diff
@@ -32,7 +32,7 @@ func GenerateFlow(password, output, identifier string, bits int) {
 		panic(fmt.Sprintf("Error creating key: %s\n", err))
 	}
 
-	fingerPrint, _ := remote_signer.GetFingerPrintFromKey(key)
+	fingerPrint, _ := tools.GetFingerPrintFromKey(key)
```
Comment (silviolleite) [inline cmd/cli/generateKey.go:35]: The same problem of previous comment about error handling.

File: cmd/cli/cmdimport.go (hunk 1 above)
Comment (racerxdl) [reply cmd/cli/cmdimport.go:64]: Don't need. The key is parsed before that.

File: cmd/cli/generateKey.go (hunk 2 above)
Comment (racerxdl) [reply cmd/cli/generateKey.go:35]: Same here, the key is guaranteed to be valid. So fingerprint will always return.

Comment (silviolleite) [reply cmd/cli/generateKey.go:35]: Okay!

File: cmd/cli/cmdimport.go (hunk 1 above)
Comment (silviolleite) [reply cmd/cli/cmdimport.go:64]: Done!

**Pull request merged**